### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by JCorrea13**
- [x] AtomicTypes Test
- [x] CircularBuffer Test
- [x] CsvWriter Test
- [x] DebugLogger Test
- [x] ExceptionExtensions Test
- [x] ThreadWorkerBase Test
- [x] ViewModelBase Test
- [x] WindowServicesExtensions Test

### CODE REVIEW

File: test/Swan.Test/CircularBufferTest.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Swan.Test
```
Comment (geoperez) [inline test/Swan.Test/CircularBufferTest.cs]: Remove this file, the class was removed.

File: test/Swan.Test/JsonClientTest.cs [hunk 2]
```diff
@@ -29,6 +29,7 @@ public async Task WithValidParams_ReturnsTrue()
             Assert.IsNotNull(data);
             Assert.IsTrue(data.ContainsKey(AuthorizationToken));
             Assert.AreEqual(responseObj[AuthorizationToken], data[AuthorizationToken]);
+            //Assert.That(data[AuthorizationToken], Is.EqualTo(responseObj[AuthorizationToken]));
```
Comment (geoperez) [inline test/Swan.Test/JsonClientTest.cs]: ??

File: test/Swan.Test/CircularBufferTest.cs (hunk 1 above)
Comment (JCorrea13) [reply test/Swan.Test/CircularBufferTest.cs]: removed

File: test/Swan.Test/JsonClientTest.cs (hunk 2 above)
Comment (JCorrea13) [reply test/Swan.Test/JsonClientTest.cs]: done

File: test/Swan.Test/CsvWriterTest.cs [hunk 3]
```diff
@@ -271,5 +271,79 @@ public void WritingHeadersFromSampleClass_WritesHeaders()
...
+            {
+                "Id", "AlternateId", "Name", "Description", "IsValidated", "ValidationResult", "Score", "CreationDate",
+                "AccessDate",
+            };
+
+            var stringHeadersOutput = string.Join(",", stringHeaders);
+
+            var objHeaders = new SampleCsvRecord();
+
+            using (var stream = new MemoryStream())
```
Comment (geoperez) [inline test/Swan.Test/CsvWriterTest.cs]: Use new using var from C# 8

**Comment by geoperez** [comment]
You need to check Ubuntu AppVeyor VM, it's throwing some errors.

**Comment by geoperez** [comment]
@JCorrea13 will you finish this PR? or do you talk with Ana?

**Comment by geoperez** [comment]
Status?

**Pull request merged**